core/templates/dev/head/pages/exploration_editor/statistics_tab/StatisticsTab.js

```diff
@@ -76,49 +89,76 @@ oppia.controller('StatisticsTab', [
-
-        if (numVisits > 0) {
-          $scope.hasExplorationBeenVisited = true;
+        if (!$scope.enableNewFramework) {
+          var data = response.data;
+          var numVisits = data.num_starts;
+          var numCompletions = data.num_completions;
+          var improvements = data.improvements;
+          $scope.stateStats = data.state_stats;
+          $scope.lastUpdated = data.last_updated;
+
+          if (numVisits > 0) {
+            $scope.hasExplorationBeenVisited = true;
+          }
+
+          $scope.chartData = [
+            ['', 'Completions', 'Non-completions'],
+            ['', numCompletions, numVisits - numCompletions]
+          ];
+
+          $scope.statsGraphOpacities = {
+            legend: 'Students entering state'
+          };
+          // TODO(bhenning): before, there was a special opacity computed for
+          // the ending (numCompletions/numVisits), should we do this for all
+          // terminal nodes, instead? If so, explorationStatesService needs to
+          // be able to provide whether given states are terminal
+
+          explorationStatesService.getStateNames().forEach(function(stateName) {
+            var visits = 0;
+            if ($scope.stateStats.hasOwnProperty(stateName)) {
+              visits = $scope.stateStats[stateName].first_entry_count;
+            }
+            $scope.statsGraphOpacities[stateName] = Math.max(
+              visits / numVisits, 0.05);
+          });
+
+          $scope.hasTabLoaded = true;
         }
 
-        $scope.chartData = [
-          ['', 'Completions', 'Non-completions'],
-          ['', numCompletions, numVisits - numCompletions]
-        ];
-
-        $scope.statsGraphOpacities = {
-          legend: 'Students entering state'
-        };
-        // TODO(bhenning): before, there was a special opacity computed for the
-        // ending (numCompletions/numVisits), should we do this for all
-        // terminal nodes, instead? If so, explorationStatesService needs to be
-        // able to provide whether given states are terminal
-
-        explorationStatesService.getStateNames().forEach(function(stateName) {
-          var visits = 0;
-          if ($scope.stateStats.hasOwnProperty(stateName)) {
-            visits = $scope.stateStats[stateName].first_entry_count;
+        // From this point on, all computation done is for the new stats
+        // framework under a development flag.
+        if ($scope.enableNewFramework) {
+          var data = response.data;
+          console.log(data);
+          var numStarts = data.num_starts_v1 + data.num_starts_v2;
+          var numActualStarts = (
+            data.num_actual_starts_v1 + data.num_actual_starts_v2);
+          var numCompletions = (
+            data.num_completions_v1 + data.num_completions_v2);
+          $scope.stateStats = data.state_stats_mapping;
+
+          if (numStarts > 0) {
+            $scope.hasExplorationBeenVisited = true;
```

Comment:
$scope.explorationHasBeenVisited (for non-functions try not to start the name with a verb).

Reply:
Done.